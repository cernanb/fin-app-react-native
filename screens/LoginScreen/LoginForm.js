import React, { Component } from 'react'
import {
  Button,
  StyleSheet,
  Text,
  View,
  TextInput,
  ScrollView
} from 'react-native'
import { Field, reduxForm } from 'redux-form'

import FormInput from '../../components/FormInput'

const validate = values => {
  const errors = {}

  if (!values.username) {
    errors.username = "Username must be filled in!"
  }

  if (!values.password) {
    errors.password = "Passworrd must be filled in!"
  } else if (values.password.length < 8) {
    errors.password = "Password must be at least 8 characters in length!"
  }
  console.log("errors: ", errors)

  return errors
}

class LoginForm extends Component {

  submit = formData => this.props.onSubmit(formData)

  render() {
    const { handleSubmit, submitting } = this.props

    return (
      <View style={styles.container}>
        <Field 
          label="Username"
          name="username"
          secureTextEntry={false}
          component={FormInput} />
        <Field 
          label="Password"
          secureTextEntry={true}
          name="password"
          component={FormInput} />
        <Button
          title="Login"
          disabled={submitting}
          onPress={handleSubmit(this.submit)}
        />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    padding: 20,
  }
})

export default reduxForm({
  form: 'login',
  validate
})(LoginForm)