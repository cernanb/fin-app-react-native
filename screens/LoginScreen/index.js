import React, { Component } from 'react'
import {
  Button,
  View, 
  Text,
  StyleSheet
} from 'react-native'
import { connect } from 'react-redux'

import LoginForm from './LoginForm'
import { login } from '../../redux/modules/Auth/actions'

class LoginScreen extends Component {
  static navigationOptions = {
    title: "Login"
  }

  handleLogin = formData => this.props.login(formData, this.props.navigation)

  render() {
    return(
      <View style={styles.container}>
        <Text>Login</Text>
        <LoginForm onSubmit={this.handleLogin} />
        <Button 
          title="Create new account"
          onPress={() => console.log('got to signup page')}
        />
      </View>
    )
  }
}

export default connect(null, { login })(LoginScreen)

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#3498db',
    alignItems: 'center',
    justifyContent: 'center',
  },
});