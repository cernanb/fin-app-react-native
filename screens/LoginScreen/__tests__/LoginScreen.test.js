import React from 'react'
import LoginScreen from '../../LoginScreen'

import renderer from 'react-test-renderer';

describe('LoginScreen', () => {
    it('renders', () => {
        const tree = renderer.create(
            <LoginScreen />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})