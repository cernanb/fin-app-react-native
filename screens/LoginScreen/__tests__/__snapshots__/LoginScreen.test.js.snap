// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#595856",
            "fontFamily": "Verdana",
            "fontSize": 20,
            "marginBottom": 10,
          }
        }
      >
        Username
      </Text>
      <TextInput
        style={
          Object {
            "backgroundColor": "#FFF",
            "fontSize": 30,
            "height": 80,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#595856",
            "fontFamily": "Verdana",
            "fontSize": 20,
            "marginBottom": 10,
          }
        }
      >
        Password
      </Text>
      <TextInput
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "#FFF",
            "fontSize": 30,
            "height": 80,
          }
        }
      />
    </View>
  </View>
</View>
`;
